locals.tf

```diff
@@ -0,0 +1,133 @@
+
+locals {
+  helm_values = [{
+    argo-cd = {
+      crds = {
+        install = false # already done during bootstrap
+      }
+      configs = merge(length(var.repositories) > 0 ? {
+        repositories = var.repositories
+        } : null, {
+        rbac = {
+          scopes           = "[groups, cognito:groups, roles]"
+          "policy.default" = ""
+          "policy.csv"     = <<-EOT
+                              g, pipeline, role:admin
+                              g, argocd-admin, role:admin
+                            EOT
```

Comment:
Dropping pipeline changes things a lot, pipeline configuration (creating secrets, token, etc), meaning that a lot of code lands in userland instead of inside the module. This auth config is widely used in most -if not all- our deployments I think this is one of the "c2c practises" that we need to materialize in the module

Reply:
I'm not suggesting to drop `pipeline` group or to create it outside the module. In your module call (below), you override the whole block and add `pipeline` and `argocd-admin` groups again, when they have already been added inside the module. I was thinking about some refact that would allow merging two `policy.csv`s, that way only new groups and policies are put in the rbac config. But again, this is something that could be maybe managed in another release.
```
helm_values = [{
    argo-cd = {
      configs = {
        rbac = {
          "scopes"         = "[groups, roles]"
          "policy.default" = ""
          "policy.csv"     = <<-EOT
            g, pipeline, role:admin
            g, argocd-admin, role:admin
            g, "${data.azuread_group.<my_department_az_group>.object_id}", role:admin
            EOT
        }
      }
    }
  }]
```

Reply:
As I said, this can wait the next release. Resolving this now.